crates/printer/src/util.rs

```diff
@@ -102,6 +106,18 @@ impl<M: Matcher> Replacer<M> {
                     },
                 )
                 .map_err(io::Error::error_message)?;
+
+            if is_multi_line {
+                // Remove the subject buffer beyond the range.end.
+                // NOTE: could this be a bug with the current replace functionality?
+                // As an example, running `rg -U '\.\nA' --replace 'BB' -C 2` in this repo produces
+                // spurious extra lines of output in the replacement, as the extra bytes where
+                // not removed. This seems to fix that, but it's not pretty and might be wrong.
+                dst.truncate(dst.len() - extension_len);
+            } else {
+                // Restore the line terminator.
```

Comment:
This is probably wrong, but I'm relying on the original line terminators instead of reproducing it in the printer. I'm not sure how to fix it properly.

Reply:
At least all tests seem to pass.